src/lib.rs

```diff
@@ -3,27 +3,264 @@
 #![cfg_attr(not(feature = "std"), no_std)]
 extern crate alloc;
 
-mod pointer;
-pub use pointer::*;
-mod token;
-pub use token::*;
-pub mod index;
-pub use index::Index;
-
-mod error;
-pub use error::*;
+use core::{fmt, num::ParseIntError};
 
+// TODO(chanced): make these pub?
 mod assign;
-pub use assign::*;
+pub use assign::{Assign, AssignError, Assignment};
 mod delete;
-pub use delete::*;
+pub use delete::Delete;
 mod resolve;
-pub use resolve::*;
+pub use resolve::{Resolve, ResolveError, ResolveMut};
 
 pub mod prelude;
 
+// TODO(chanced): move these into lib.rs
```

Comment:
Wait, not sure I follow this. Exporting symbols from these in `lib.rs` is fine, but I think it's an anti-pattern to dump everything public here. If you are trying to get rid of the glob uses, I think it's best to just enumerate the symbols as you've done above.

Reply:
Oh, those were from before the comment above about restructuring the surface.